File: leetcode/python/0441_arranging_coins.py
class Solution:
    def arrangeCoins(self, n: int) -> int:
        # coins = 1
        # rows = 0
        # while n > 0:
        # n -= coins
        # if n >= 0: rows += 1
        # coins += 1
        # return rows
        coins, rows = 1, 0
        while n > 0:
            n -= coins
            if n >= 0: rows += 1
            coins += 1

        return rows

File: leetcode/java/0441_arranging_coins.java
class Solution {
    public int arrangeCoins(int n) {
        // n = 5: 1, 2, 3 -> (n - i (i goes from 1...inf while n >= i))
        // if n >= i: res++
        // return res 
        int i = 1, res = 0;
        while (n >= i) {
            if (n >= i) res++;
            n -= i;
            i++;
        }

        return res;
    }
}